de.tudarmstadt.ukp.wikipedia.parser/src/test/java/de/tudarmstadt/ukp/wikipedia/parser/ParserTest.java

```diff
@@ -36,4 +39,49 @@ public void testGalleryLinks(){
 
 
     }
+
+    public boolean findLinkInList(Link l, Collection<Link> list){
+         for(Link currentLink:list){
+             if (l.equals(currentLink)) return true;
+         }
+        return false;
+    }
+
+
+    @Test
+    public void testExtractingLinksFromGallery(){
+
+        String text = "==History==\n" +
+                "<gallery>\n" +
+                "Image:Buckingham Branch Railroad GP16 rebuild.JPG|BB 2, a [[GP16]], getting a new coat of paint at [[Dillwyn, Virginia]].\n" +
+                "Image:BBRR4 Dillwyn WJGrimes.JPG|BBRR 4, an [[RS-4-TC]], with a new coat of paint at [[Dillwyn, Virginia]].\n" +
+                "Image:BB7 Louisa WJGrimes.JPG|BB 7, a [[GP40]], heading east at [[Louisa, Virginia]].\n" +
+                "Image:BBRR8 Doswell WJGrimes.JPG|BB 8, a GP16 in GRIV paint at [[Doswell, Virginia]].\n" +
+                "Image:Bb 13 augusta co-op.jpg|BB 13 switches Augusta CO-OP in [[Staunton, Virginia]].\n" +
+                "Image:Bb_7_svrr_staunton,_va_08292010.jpg|BB 7 with sisters, south on the [[Shenandoah Valley Railroad (short-line)|Shenandoah Valley Railroad]] in [[Staunton, Virginia]]. Having just made a pick up of Empty Cars to take West.\n" +
+                "Image:image_with_no_par.jpg\n" +
+                "Image:image_with_par.jpg|Something [[Spiderman]]\n" +
+                "</gallery>";
+
+        MediaWikiParserFactory pf = new MediaWikiParserFactory(WikiConstants.Language.english);
+        MediaWikiParser parser = pf.createParser();
+        ParsedPage pp = parser.parse(text);
+        Paragraph testPar;
+        testPar = pp.getSections().get(0).getParagraphs().get(0);
+
+        assert(this.findLinkInList(new Link(testPar, new Span(8, 12), "GP16",  Link.type.INTERNAL, new ArrayList<String>()), testPar.getLinks()));
+        assert(this.findLinkInList(new Link(testPar, new Span(45, 62), "Dillwyn,_Virginia",  Link.type.INTERNAL, new ArrayList<String>()), testPar.getLinks()));
+        assert(this.findLinkInList(new Link(testPar, new Span(312, 338), "Shenandoah_Valley_Railroad_(short-line)",  Link.type.INTERNAL, new ArrayList<String>()), testPar.getLinks()));
```

Comment:
Im not checking the Link.type in the equals function. Just selected `Internal` as a type was requiered for the constructor. 
-  `Internal` type in this case means that is a link to some other article in the wiki.
- `external` type a link outside the wiki
- `audio`, `video` I guess these are format dependent.